### PR TITLE
Add nucleotide-based genome clustering to pipeline

### LINKED ISSUE #7: Add aniclust and anicalc modules to workflow
### Description of feature

- [ ] Add script to `bin/anicalc.py` and `aniclust.py` for calculating ANI and clustering based on ANI using blastn output
- [ ] Create anicalc and aniclust modules using `nf-core modules create` and by modifying code
- [ ] Add anicalc and aniclust modules to `workflows/phageannotator.nf`
- [ ] Update `nextflow.params` to include anicalc and aniclust parameters
- [ ] Update `nextflow_schema.json` to include anicalc and aniclust params using `nf-core schema build`
- [ ] Update `conf/modules.config` to include anicalc and aniclust extra arguments
- [ ] Test anicalc and aniclust modules using `nextflow run . -profile test,singularity --outdir results`
- [ ] Lint additions using `nf-core lint`
- [ ] Sync anicalc and aniclust branch with nf-core/phageannotater:dev once PR #6  has been merged.
- [ ] Submit a PR for anicalc and aniclust modules

### PR DESCRIPTION
**Pull request by CarsonJM**
## PR checklist

- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/phageannotator/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/phageannotator _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [X] Make sure your code lints (`nf-core lint`).
- [X] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by maxulysse — APPROVED** [review]
LGTM

File: modules/local/anicluster/anicalc/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,32 @@
+nextflow_process {
+
+    name "Test process: ANICLUSTER_ANICALC"
+    script "modules/local/anicluster/anicalc/main.nf"
```
Comment (sateeshperi) [inline modules/local/anicluster/anicalc/tests/main.nf.test:4]:
```suggestion
    script "../main.nf"
```

File: modules/local/anicluster/anicalc/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    process "ANICLUSTER_ANICALC"
+    tag "modules"
+    tag "modules_local"
+    tag "anicluster"
+    tag "anicluster_anicalc"
+
+
+    test("Should run without failures") {
```
Comment (sateeshperi) [inline modules/local/anicluster/anicalc/tests/main.nf.test:12]:
```suggestion
    test("modules_local blast_txt") {
```

File: modules/local/anicluster/aniclust/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,33 @@
+nextflow_process {
+
+    name "Test process: ANICLUSTER_ANICLUST"
+    script "modules/local/anicluster/aniclust/main.nf"
```
Comment (sateeshperi) [inline modules/local/anicluster/aniclust/tests/main.nf.test:4]:
```suggestion
    script "../main.nf"
```

File: modules/local/anicluster/aniclust/tests/main.nf.test [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    process "ANICLUSTER_ANICLUST"
+    tag "modules"
+    tag "modules_local"
+    tag "anicluster"
+    tag "anicluster_aniclust"
+
+
+    test("Should run without failures") {
```
Comment (sateeshperi) [inline modules/local/anicluster/aniclust/tests/main.nf.test:12]: more descriptive test name

File: modules/local/anicluster/aniclust/tests/main.nf.test [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+        when {
+            params {
+                outdir   = "$outputDir"
+            }
+            process {
+                """
+                input[0] = [ [ id: 'test' ], [ file('https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists: true ) ],
```
Comment (sateeshperi) [inline modules/local/anicluster/aniclust/tests/main.nf.test:20]: can this illumina fasta be added to test_data.config and referred from there ?

File: modules/local/anicluster/extractreps/tests/main.nf.test [hunk 6]
```diff
@@ -0,0 +1,33 @@
+nextflow_process {
+
+    name "Test process: ANICLUSTER_EXTRACTREPS"
+    script "modules/local/anicluster/extractreps/main.nf"
```
Comment (sateeshperi) [inline modules/local/anicluster/extractreps/tests/main.nf.test:4]:
```suggestion
    script "../main.nf"
```

File: subworkflows/local/fasta_all_v_all_blast/tests/main.nf.test [hunk 7]
```diff
@@ -0,0 +1,32 @@
+nextflow_workflow {
+
+    name "Test Subworkflow: FASTA_ALL_V_ALL_BLAST"
+    script "subworkflows/local/fasta_all_v_all_blast/main.nf"
```
Comment (sateeshperi) [inline subworkflows/local/fasta_all_v_all_blast/tests/main.nf.test:4]: same comments as above

**Review by sateeshperi — APPROVED** [review]
few minor comments...totally ok if you want to roll them here or future ones

**Comment by CarsonJM** [comment]
Thanks a bunch @sateeshperi and @maxulysse! @sateeshperi I will standardize the nf.test files according to your recommendations in the upcoming PR!

**Pull request merged**